src/components/TodoItem.tsx

```diff
@@ -0,0 +1,853 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { TodoItemProps, Subtask, Checklist } from '@/types/todo';
+import { calculateSubtaskProgress, calculateChecklistProgress, getPriorityColor, formatDueDate, createDefaultTodo, calculateTodoComplexity } from '@/utils/todoUtils';
+import SubtaskItem from './SubtaskItem';
+import SubtaskForm from './SubtaskForm';
+import ChecklistItem from './ChecklistItem';
+import ChecklistForm from './ChecklistForm';
+
+export default function TodoItem({ todo, onToggle, onDelete }: TodoItemProps) {
+  const [showSubtasks, setShowSubtasks] = useState(false);
+  const [showChecklists, setShowChecklists] = useState(false);
+  const [showSubtaskForm, setShowSubtaskForm] = useState(false);
+  const [showChecklistForm, setShowChecklistForm] = useState(false);
+  const [expanded, setExpanded] = useState(false);
+  const [hovered, setHovered] = useState(false);
+  const [selected, setSelected] = useState(false);
+  const [animationState, setAnimationState] = useState('idle');
+  const [lastUpdate, setLastUpdate] = useState(new Date());
+  const [showDetails, setShowDetails] = useState(false);
+  const [subtasks, setSubtasks] = useState<Subtask[]>(todo.subtasks || []);
+  const [checklists, setChecklists] = useState<Checklist[]>(todo.checklists || []);
+  const [subtaskProgress, setSubtaskProgress] = useState(0);
+  const [checklistProgress, setChecklistProgress] = useState(0);
+  const [totalSubtasks, setTotalSubtasks] = useState(0);
+  const [completedSubtasks, setCompletedSubtasks] = useState(0);
+  const [totalChecklistItems, setTotalChecklistItems] = useState(0);
+  const [completedChecklistItems, setCompletedChecklistItems] = useState(0);
+  const [hasSubtasks, setHasSubtasks] = useState(false);
+  const [hasChecklists, setHasChecklists] = useState(false);
+  const [priority, setPriority] = useState(todo.priority || 3);
```

Comment:
[MEDIUM] Hardcoded priority value 3 used without constant.

**Code Smell** detected

Suggested change:

```suggestion
const DEFAULT_PRIORITY = 3;
const [priority, setPriority] = useState(DEFAULT_PRIORITY);
```